### PR TITLE
Add simple query variables code generation

### LINKED ISSUE #41: [Code Generation] Generate query wrapper class around query text source and variables
Create generic GraphQLQuery interface like iOS does
Generate query class implemtentation for each query by implementing GraphQLQuery.
Generated query class implemtentations should allow set arguments.

See https://github.com/apollostack/apollo-android/blob/design-doc/DESIGN.md#query-classes
Swift examples can be found here: https://github.com/apollostack/apollo-ios/blob/master/Tests/ApolloTests/StarWars/API.swift

### PR DESCRIPTION
**Pull request by sav007**
Add code generation to support query variables.
Update unit tests

Example of generated query with variables: https://github.com/apollostack/apollo-android/blob/2aaa89c1b3a466f85fe33d01582bc348468566eb/apollo-compiler/src/test/graphql/com/example/simple_arguments/HeroNameConditionalInclusionExpected.java

Example of usage:
```
new Query(
        new Query.Variables.Builder()
            .episode(Episode.EMPIRE)
            .includeName(true)
            .build()
    );
```

Closes #41 

@felipecsl @martijnwalraven pls take a look

### CODE REVIEW

**Comment by felipecsl** [comment]
Why use a map? Since you know all the variable names and types at compile time, you should be able to generate fields for them, so you don't need the map at all!

**Comment by sav007** [comment]
The map used as convenient to make serialization easier. Query executor to send request requires 3 things: query definition, fragment definitions and variables. So by using map it won't require to have custom retrofit converter to serialize variables to json. This map is supposed to be used internally for that reason. Otherwise we will require to have json convert like moshi, gson etc.

**Comment by sav007** [comment]
And I guess the same reason on iOS they used map for variables

**Comment by sav007** [comment]
@felipecsl oh Felipe, Felipe you made my PR to have tons of conflicts :)

**Comment by felipecsl** [comment]
@sav007 hah so sorry man!! 🙈

**Comment by felipecsl** [comment]
hmm ok i'm not super convinced about using a Map yet, but that's fine for now

File: apollo-compiler/src/main/kotlin/com/apollostack/compiler/ClassNames.kt [hunk 1]
```diff
@@ -10,4 +12,16 @@ object ClassNames {
   val COLLECTIONS: ClassName = ClassName.get(Collections::class.java)
   val ARRAYS: ClassName = ClassName.get(Arrays::class.java)
   val QUERY: ClassName = ClassName.get(Query::class.java)
+  val OBJECT: ClassName = ClassName.get(Object::class.java)
+  val MAP: ClassName = ClassName.get(Map::class.java)
+  val HASH_MAP: ClassName = ClassName.get(HashMap::class.java)
+
+  fun parameterizedMap(firstTypeArgument: TypeName, secondTypeArgument: TypeName): TypeName =
```
Comment (felipecsl) [inline apollo-compiler/src/main/kotlin/com/apollostack/compiler/ClassNames.kt]: nit: what about appending `of` at the end to match the other Kotlin factory methods? eg.: `parameterizedMapOf()`

File: apollo-compiler/src/main/kotlin/com/apollostack/compiler/QueryTypeSpecBuilder.kt [hunk 2]
```diff
@@ -73,19 +73,61 @@ class QueryTypeSpecBuilder(
...
     return this
   }
 
+  private fun TypeSpec.Builder.addVariablesDefinitions(variables: List<Variable>): TypeSpec.Builder {
+    if (variables.isEmpty()) {
+      addMethod(MethodSpec.methodBuilder(VARIABLE_DEFINITIONS_ACCESSOR_NAME)
+          .addAnnotation(Annotations.OVERRIDE)
+          .addModifiers(Modifier.PUBLIC)
+          .returns(ClassNames.parameterizedMap(ClassNames.STRING, ClassNames.OBJECT))
+          .addStatement("return \$T.EMPTY_MAP", ClassNames.COLLECTIONS)
```
Comment (felipecsl) [inline apollo-compiler/src/main/kotlin/com/apollostack/compiler/QueryTypeSpecBuilder.kt]: `Collections.emptyMap()` is slightly better

File: apollo-compiler/src/test/graphql/com/example/simple_arguments/TestQuery.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
+{
+	"operations": [
```
Comment (felipecsl) [inline apollo-compiler/src/test/graphql/com/example/simple_arguments/TestQuery.json]: mind formatting this file to use 2-space indentation like the other files?

**Comment by sav007** [comment]
@felipecsl any other comments, merge?

**Pull request merged**